aws-android-sdk-s3/src/test/java/com/amazonaws/mobileconnectors/s3/transferutility/UploadInputStreamTest.java

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.mobileconnectors.s3.transferutility;
+
+import android.content.Context;
+import android.util.Log;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.UUID;
+
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.isNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(RobolectricTestRunner.class)
+public class UploadInputStreamTest {
+
+    private static final String TAG = "InputStreamTest";
+
+    private static final String DEFAULT_BUCKET = "s3-upload-input-stream-unit-test-" + getRandomString();
+    private final InputStream inputStream;
+    private final ObjectMetadata metadata;
+    private final CannedAccessControlList cannedAcl;
+    private final TransferListener listener;
+    private final TransferUtility transferUtility;
+
+    public UploadInputStreamTest() {
+        AmazonS3Client s3 = mock(AmazonS3Client.class);
```

Comment:
Rather than setting all of this up in a constructor, it's better to put it in a setup method with the @Before annotation, like this:
```
@Before
public void setup() {
    AmazonS3Client s3 = mock(AmazonS3Client.class);
    ....
}
```

With this approach, everything in the `@Before` method will be run before each individual test is run.  This allows you to reset / reinitialize everything to a default state, so that one test won't cause side effects on another test.  This might also fix the `IllegalStateException` that I see most of these tests are throwing right now. 

I'd also suggest creating a teardown method with the `@After` annotation like this:
```
@After
public void teardown() {
    inputStream.close();
}
```